tv_apps/smart-home/js/home.js

```diff
@@ -187,10 +195,148 @@
+
+      this.renameMenuItem.addEventListener('click', () => {
+        this.enterEditMode()
+          .then(() => {
+            var focus = this.spatialNavigator.getFocusedElement();
+            return this.edit.renameCard(focus,
+              focus.getNodeFromItem(focus.currentItem));
+          })
+          .then(() => this.exitEditMode());
+      });
+
+      this.removeMenuItem.addEventListener('click', () => {
+        this.enterEditMode()
+          .then(() => {
+            var focus = this.spatialNavigator.getFocusedElement();
+            return this.edit.openDeleteCardDialog(focus,
+              focus.getNodeFromItem(focus.currentItem));
+          })
+          .then(result => {
+            if (result.agreeToDelete) {
+              // We should wait until card is removed then exit edit mode.
+              // This is to avoid messed with card removing transition.
+              if (result.scrollable === this.folderScrollable) {
+                var folder = this.cardManager.findContainingFolder(
+                    { cardId: result.scrollable.currentItem.dataset.cardId });
+                folder.once('card-removed', () => this.exitEditMode());
+              } else {
+                this.cardManager.once('card-removed',
+                  () => this.exitEditMode());
+              }
+            } else {
+              // Don't agree to delete. Nothing happens so let's just exit.
+              this.exitEditMode();
+            }
+          });
+      });
+
+      this.editFolderMenuItem.addEventListener('click', () => {
+        this._cardPicker.show(this.cardScrollable.currentItem);
+      });
+    },
+
+    enableContextMenu: function () {
+      var focus = this.spatialNavigator.getFocusedElement();
+      if (this.mode === '' &&
+          (focus === this.cardScrollable || focus === this.folderScrollable)) {
+        this.mainSection.setAttribute('contextmenu', this.contextmenuElem.id);
+      }
+    },
+
+    disableContextMenu: function () {
+      this.mainSection.setAttribute('contextmenu', '');
+    },
+
+    updateContextMenu: function (itemElem) {
+      var type = itemElem.getAttribute('app-type');
+      if (this.contextmenuElem.dataset.currentAppType == type) {
+        return;
+      }
+      this.contextmenuElem.dataset.currentAppType = type;
+
+      // Clean up first
+      for (var i = this.contextmenuElem.childNodes.length - 1; i >= 0; --i) {
+        this.contextmenuElem.removeChild(this.contextmenuElem.childNodes[i]);
+      }
+
+      this.contextmenuElem.appendChild(this.moveMenuItem);
+
+      switch (type) {
+        case 'folder':
+          this.contextmenuElem.appendChild(this.editFolderMenuItem);
+          break;
+
+        case 'tv':
+        case 'app':
+        case 'appbookmark':
+          this.contextmenuElem.appendChild(this.renameMenuItem);
+          this.contextmenuElem.appendChild(this.removeMenuItem);
+          break;
+      }
+    },
+
+    enterEditMode: function (onEnter) {
```

Comment:
Actually I don't think adding a pair of enter/edit mode in home.js is a good idea.